vllm/model_executor/layers/fused_moe/layer.py

```diff
@@ -579,9 +631,12 @@ def select_experts(hidden_states: torch.Tensor,
         return topk_weights, topk_ids
 
     def forward(self, hidden_states: torch.Tensor,
-                router_logits: torch.Tensor):
+                router_logits: torch.Tensor) -> torch.Tensor:
         assert self.quant_method is not None
-
+        
+        if self.expert_map is not None:
+            self.expert_map = self.expert_map.to(hidden_states.device)
```

Comment:
can we directly put this into `__init__` with the correct device? doing it in `forward` might break `torch.compile`